packages/core/src/core.ts

```diff
@@ -419,7 +423,7 @@ class ThresholdKey implements ITKey {
     }
     const privKey = lagrangeInterpolation(shareArr, shareIndexArr);
     // check that priv key regenerated is correct
-    const reconstructedPubKey = getPubKeyPoint(privKey);
+    const reconstructedPubKey = getPubKeyPoint(privKey, this.keyType);
```

Comment:
getPubKeyPoint should use `Point` class
we can add static function fromPrivateKey

Reply:
done